auth-service/src/main/kotlin/uk/gov/communities/delta/auth/repositories/LdapRepository.kt

```diff
@@ -243,26 +238,22 @@ class LdapRepository(
         }
     }
 
-    fun createUserWithRoles(userAttrs: Attributes, allGroups: List<String>, organisationId: String): UserWithRoles {
-        val userRoles = allGroups.filter {
-            it.contains("${datamartDeltaDataProviders}$organisationId", ignoreCase = true) ||
-                it.contains("${datamartDeltaDataCertifiers}$organisationId", ignoreCase = true)
-        }.map { role ->
+    private fun createUserWithRoles(user: LdapUser, organisationId: String): UserWithRoles {
+        val allGroups = user.memberOfCNs
+        val userRoles = allGroups.mapNotNull { role ->
             when {
-                role.contains("${datamartDeltaDataProviders}$organisationId", ignoreCase = true) -> "Data provider"
-                role.contains("${datamartDeltaDataCertifiers}$organisationId", ignoreCase = true) -> "Data certifier"
-                else -> role
+                role.equals("${datamartDeltaDataProviders}$organisationId", ignoreCase = true) -> "Data provider"
+                role.equals("${datamartDeltaDataCertifiers}$organisationId", ignoreCase = true) -> "Data certifier"
+                role.equals("${datamartDeltaStatsDataCertifiers}$organisationId", ignoreCase = true) -> "Data certifier"
+                else -> null
```

Comment:
Just checking - are both certifier types the same? i.e. both would show as 'Data certifier'?

Reply:
They're slightly different but I don't think we need to distinguish between them here.